src/realm/query_expression.hpp

```diff
@@ -1479,27 +1543,62 @@ class Value : public ValueBase, public Subexpr2<T> {
+                if constexpr (std::is_same_v<T, Mixed>) {
+                    if (auto common_type = Mixed::get_common_type(left->m_storage[0], right->m_storage[m])) {
+                        Mixed l = left->m_storage[0].export_to(*common_type);
+                        Mixed r = right->m_storage[m].export_to(*common_type);
+                        if (c(l, r, left_is_null, r.is_null()))
+                            return m;
+                    }
+                }
+                else if (c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
                     return m;
             }
         }
         else if (comparison == ExpressionComparisonType::None) {
             for (size_t m = 0; m < sz; m++) {
-                if (c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
+                if constexpr (std::is_same_v<T, Mixed>) {
+                    if (auto common_type = Mixed::get_common_type(left->m_storage[0], right->m_storage[m])) {
+                        Mixed l = left->m_storage[0].export_to(*common_type);
+                        Mixed r = right->m_storage[m].export_to(*common_type);
+                        if (c(l, r, left_is_null, r.is_null()))
+                            return not_found;
+                    }
+                }
+                else if (c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
                     return not_found;
             }
             return 0; // no values matched, return match
         }
         else if (comparison == ExpressionComparisonType::All) {
             for (size_t m = 0; m < sz; m++) {
-                if (!c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
+                if constexpr (std::is_same_v<T, Mixed>) {
+                    if (auto common_type = Mixed::get_common_type(left->m_storage[0], right->m_storage[m])) {
+                        Mixed l = left->m_storage[0].export_to(*common_type);
+                        Mixed r = right->m_storage[m].export_to(*common_type);
+                        if (!c(l, r, left_is_null, r.is_null()))
+                            return not_found; // one did not match
+                    }
+                    else {
+                        return not_found; // one did not match
+                    }
+                }
+                else if (!c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
                     return not_found; // one did not match
             }
             return 0; // all values matched (or there this is an empty list), return match
         }
         else { // ANY from list
             for (size_t m = 0; m < sz; m++) {
-                if (c(left->m_storage[0], right->m_storage[m], left_is_null, right->m_storage.is_null(m)))
+                if constexpr (std::is_same_v<T, Mixed>) {
```

Comment:
No. The mixed compare function only returns whether the values are less, equal or greater - it can't handle the case where the types are not comparable.